OpenOversight/app/main/views.py

```diff
@@ -305,6 +305,47 @@ def list_officer(department_id, page=1, from_search=False):
         from_search=from_search)
 
 
+@main.route('/department/<int:department_id>/chart')
+def dept_org_chart(department_id):
+    department = Department.query.filter_by(id=department_id).first()
+    if not department:
+        abort(404)
+
+    officers = Officer.query.filter(Officer.department_id == department_id).all()
+    # map each officer to their most recent assignment
+    assign_dict = {}
+    assign_records = Assignment.query.all()
+    for record in assign_records:
+        if record.officer_id not in assign_dict:
+            assign_dict[record.officer_id] = record
+        else:
+            if record.star_date > assign_dict[record.officer_id].star_date:
+                assign_dict[record.officer_id] = record
+
+    officer_nodes = {}
+    rank_types = ["NOT SURE"]
+
+    for officer in officers:
+        rec = assign_dict[officer.id]
```

Comment:
It's possible there exist officers that don't have assignments, so this should probably be wrapped in a try block:
```
try:
    rec = assign_dict[officer.id]
except KeyError:
    continue
```

Reply:
thanks, i hadn't considered this case. fixed in last commit